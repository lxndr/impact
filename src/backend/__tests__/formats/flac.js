import path from 'path';
import flacParser from '../../formats/flac';

/**
 * @param {string} filename
 */
const createFile = filename => ({
  type: 'media',
  path: path.join(__dirname, filename),
  mtime: new Date(),
  size: 0,
  hash: '',
});

const scanner = {
  inspect: jest.fn(),
};

describe('FLAC parser', () => {
  it('opens file without tags', async () => {
    const file = createFile('../fixtures/audio-01.flac');
    const data = await flacParser({ file, scanner });
    expect(data).toMatchSnapshot();
  });

  it('opens file with tags', async () => {
    const file = createFile('../fixtures/audio-02.flac');
    const data = await flacParser({ file, scanner });
    expect(data).toMatchSnapshot();
  });
});
