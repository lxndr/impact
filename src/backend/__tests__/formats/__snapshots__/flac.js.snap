// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FLAC parser opens file with tags 1`] = `
Array [
  Object {
    "artist": "Artist",
    "discNumber": 1,
    "discTitle": "Disc Title",
    "releaseDate": "2018-07-07",
    "releaseType": "album",
    "title": "Album Title",
    "tracks": Array [
      Object {
        "artists": Array [
          "Artist",
        ],
        "duration": 5,
        "file": Object {
          "hash": "2d82a056c96d8998a7e1b29dc81a450d",
          "mtime": 2019-02-28T19:37:36.599Z,
          "path": "/home/lxndr/devel/research/impact/src/backend/__tests__/fixtures/audio-02.flac",
          "size": 0,
          "type": "media",
        },
        "genre": "Test Audio",
        "images": Array [],
        "nChannels": 1,
        "number": 2,
        "sampleRate": 8000,
        "title": "Track Title",
      },
    ],
  },
]
`;

exports[`FLAC parser opens file without tags 1`] = `
Array [
  Object {
    "artist": undefined,
    "discNumber": undefined,
    "discTitle": undefined,
    "releaseDate": undefined,
    "releaseType": "album",
    "title": undefined,
    "tracks": Array [
      Object {
        "artists": Array [],
        "duration": 5,
        "file": Object {
          "hash": "2d82a056c96d8998a7e1b29dc81a450d",
          "mtime": 2019-02-28T19:37:36.579Z,
          "path": "/home/lxndr/devel/research/impact/src/backend/__tests__/fixtures/audio-01.flac",
          "size": 0,
          "type": "media",
        },
        "genre": undefined,
        "images": Array [],
        "nChannels": 1,
        "number": undefined,
        "sampleRate": 8000,
        "title": undefined,
      },
    ],
  },
]
`;
